h/api/test/storage_test.py

```diff
@@ -627,6 +623,74 @@ def db(self):
         return mock.Mock(spec=db.Session)
 
 
+@pytest.mark.usefixtures('models',
+                         'update_document_metadata')
+class TestUpdateAnnotation(object):
+
+    def test_it_calls_get(self, annotation_data, models):
+        storage.update_annotation(mock.Mock(),
+                                  'test_annotation_id',
+                                  annotation_data)
+
+        models.Annotation.query.get.assert_called_once_with(
+            'test_annotation_id')
+
+    def test_it_updates_the_annotation(self, annotation_data, models):
+        annotation = models.Annotation.query.get.return_value = mock.Mock()
+
+        storage.update_annotation(mock.Mock(),
+                                  'test_annotation_id',
+                                  annotation_data)
+
+        for key, value in annotation_data.items():
+            assert getattr(annotation, key) == value
```

Comment:
Hmm, I dunno. Maybe. On the other hand this isn't the `models` package, it's a user of the models package, so I feel it ought to mock out the models package and leave it up to the models package tests to make sure the db works. All `storage` does is set the values on the model object, and that's what this tests. I could be convinced..